frontend/ui/screensaver.lua

```diff
@@ -85,7 +85,32 @@ local function _getRandomImage(dir)
     local match_func = function(file) -- images, ignore macOS resource forks
         return not util.stringStartsWith(ffiUtil.basename(file), "._") and DocumentRegistry:isImageFile(file)
     end
-    return filemanagerutil.getRandomFile(dir, match_func)
+    -- If the user has set the option to cycle images alphabetically, we sort the files instead of picking a random one.
+    if G_reader_settings:isTrue("screensaver_cycle_images_alphabetically") then
+        local files = {}
+        util.findFiles(dir, function(file)
+            if match_func(file) then
+                table.insert(files, file)
+            end
+        end, false)
+        if #files == 0 then return end
+        -- we have files, sort them in natural order, i.e z2 < z11 < z20
+        local sort = require("frontend/sort")
+        local natsort = sort.natsort_cmp()
+        table.sort(files, function(a, b)
+            return natsort(a, b)
+        end)
+        local index = G_reader_settings:readSetting("screensaver_cycle_index", 1)
+        if index > #files then -- wrap around
+            index = 1
+        end
+        local file = files[index]
+        index = (index % #files) + 1
```

Comment:
Can be just `index = index + 1` after line 103, and saving default 0 in line 103.
I.e. saving the last shown image index instead of the next image index.